src/main/java/roomescape/domain/reservation/dto/ReservationResponse.java

```diff
@@ -0,0 +1,45 @@
+package roomescape.domain.reservation.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import roomescape.domain.reservation.Reservation;
+import roomescape.domain.time.ReservationTime;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class ReservationResponse {
+    private Long id;
+    private String name;
+    private LocalDate date;
+    private Long timeId;
+    private LocalTime timeValue;
+
+    public ReservationResponse(Reservation reservation) {
+        this.id = reservation.getId();
+        this.name = reservation.getName();
+        this.date = reservation.getDate();
+        this.timeId = reservation.getTimeId();
+        this.timeValue = reservation.getTime() != null ? reservation.getTime().toLocalTime() : null;
+    }
+
+    public ReservationResponse(Reservation reservation, ReservationTime reservationTime) {
+        this(reservation);
+        if (reservationTime != null) {
+            this.timeId = reservationTime.getId();
+            this.timeValue = reservationTime.getTime();
+        }
+    }
+
+    public static ReservationResponse fromReservation(Reservation reservation) {
+        return new ReservationResponse(reservation);
+    }
+
+    public static ReservationResponse fromReservation(Reservation reservation, ReservationTime reservationTime) {
```

Comment:
[정적 팩토리 메서드(Static Factory Method)](https://tecoble.techcourse.co.kr/post/2020-05-26-static-factory-method/)의 네이밍 컨벤션에 따르면 '여러개의 매개 변수를 받아서 객체를 생성'하는 경우 of를 사용합니다
```suggestion
    public static ReservationResponse of(Reservation reservation, ReservationTime reservationTime) {
```